src/fuselayer/fuselayer.py

```diff
@@ -43,83 +42,82 @@ def chown(self, path, uid, gid):
-        '''
-        Executes the lstat call directly on the local filesystem
-            if path[0] == "/":
-            path = path[1:]
-            ret = os.lstat(newpath)
-            # print(ret)
-            return {
-                "st_atime": ret.st_atime,
-                "st_ctime": ret.st_ctime,
-                "st_gid": ret.st_gid,
-                "st_mode": ret.st_mode,
-                "st_mtime": ret.st_mtime,
-                "st_nlink": ret.st_nlink,
-                "st_size": ret.st_size,
-                "st_uid": ret.st_uid,
-            }
-        '''
-        
+     
+        local_path = os.path.join(self.rootdir, path.lstrip('/'))
         now = time.time()
         default_attrs = {
             'st_atime': now,  
             'st_ctime': now,  
             'st_mtime': now,  
             'st_gid': os.getgid(),  
             'st_uid': os.getuid(),  
-            'st_nlink': 2 if path == "/" else 1,  
+            'st_blksize': 4096, #assume size
         }
    
         if path == "/":
             attrs = {'st_mode': (S_IFDIR | 0o755), **default_attrs} 
             return attrs
 
-        for item in self.metadata:
-            if item["path_lower"] == path:
-                if item["type"] == "folder":
-                    attrs = {'st_mode': (S_IFDIR | 0o755), **default_attrs}
-                else:  # type == "file"
-                    attrs = {'st_mode': (S_IFREG | 0o644), 'st_size': item["size"], **default_attrs} 
-                return attrs
-
+        if os.path.exists(local_path):
+            try:
+                ret = os.stat(local_path)
+                return {
+                    "st_atime": ret.st_atime,
+                    "st_ctime": ret.st_ctime,
+                    "st_gid": ret.st_gid,
+                    "st_mode": ret.st_mode,
+                    "st_mtime": ret.st_mtime,
+                    "st_nlink": ret.st_nlink,
+                    "st_size": ret.st_size,
+                    "st_uid": ret.st_uid,
+                
+                }
+            except OSError as e:
+                raise FuseOSError(e.errno)
+            
+        if path in self.metadata:
+            m_data = self.metadata[path]
+            attrs = {
+                **default_attrs,
+                'st_mode': (S_IFDIR | 0o755) if m_data[2] == "folder" else (S_IFREG | 0o644),
+                'st_nlink': 2 if m_data[2] == "folder" else 1,
+                'st_size': m_data[1] if m_data[1] is not None else 0,
+                
+            }
+            st_size = m_data[1] if m_data[1] is not None else 0
+            attrs['st_blocks'] = (st_size + 511) // 512
+            return attrs
+        
         # If the path does not exist in the metadata
         raise FuseOSError(errno.ENOENT)
         
 
     def getxattr(self, path, name, position=0):
         logger.info(f"GETXATTR CALLED WITH ID {random.randint(0, 100)}")
-        
+        ret = b'default'
```

Comment:
what is this???

Reply:
This is used to prevent errors when using the `ls -l` command. The `ls -l` command calls `os.getxattr` to obtain the extended attribute; if the file hasn't been downloaded, `ls -l` will return `errno.ENOENT`. Generally, the extension attribute is a string reflecting information about the file, such as "backup" or "image/jpeg". Therefore, I am considering setting its extension attribute to "default" when the file does not exist. After it's downloaded, `getxattr` will reflect the true extension information.